test/suites/standard2/workload-partitioning.robot

```diff
@@ -1,21 +1,122 @@
 
+Resource            ../../resources/microshift-config.resource
 Resource            ../../resources/common.resource
+Resource            ../../resources/systemd.resource
 Resource            ../../resources/microshift-process.resource
 Resource            ../../resources/microshift-network.resource
 
 Suite Setup         Setup Suite With Namespace
 Suite Teardown      Teardown Suite With Namespace
 
+*** Variables ***
+${CPUsControlPlanes}    2
+${SystemdCrioDropIn}    /etc/systemd/system/crio.service.d/microshift-cpuaffinity.conf
+${SystemdMicroshiftDropIn}    /etc/systemd/system/microshift.service.d/microshift-cpuaffinity.conf
+${crioConfigFile}    /etc/crio/crio.conf.d/20-microshift-wp.conf
 
 *** Test Cases ***
 Control Plane Pods Must Be Annotated
-    [Documentation]    Verify that all the Control Plane pods are properly annotated.
-    All Pods Should Be Annotated As Management
+   [Documentation]    Verify that all the Control Plane pods are properly annotated.
+   All Pods Should Be Annotated As Management
 
+Workload Partitioning Should Work
+    [Documentation]    Verify that all the Control Plane pods are properly annotated.
+    Configure Kubelet For Workload Partitioning    ${CPUsControlPlanes}    
+    Configure crio For Workload Partitioning    ${CPUsControlPlanes} 
+    Configure CPUAffinity In Systemd    ${CPUsControlPlanes}    ${SystemdCrioDropIn}
+    Configure CPUAffinity In Systemd    ${CPUsControlPlanes}    ${SystemdMicroshiftDropIn}
+    Systemctl    restart    crio.service
+    Restart MicroShift
+    Microshift Services Should be Running On Reserved CPU    ${CPUsControlPlanes}
+    All Management Pods Should Run On Reserved CPU    ${CPUsControlPlanes}
+#TODO: cleanup,undo all the configuration
+#TODO: add workloads and check their CPU
 
 *** Keywords ***
+
+Configure Kubelet For Workload Partitioning
+    [Documentation]    configure microshift with kubelet CPU configuration
+    [Arguments]    ${cpus}
+
+    ${kubelet_config}=    CATENATE    SEPARATOR=\n
+    ...    apiServer:
+    ...    \ \ subjectAltNames:
+    ...    \ \ -\ ${USHIFT_HOST}
+    ...    kubelet:
+    ...    \ \ reservedSystemCPUs: "${cpus}"
+    ...    \ \ cpuManagerPolicy: static
+    ...    \ \ cpuManagerPolicyOptions:
+    ...    \ \ \ \ full-pcpus-only: "true"
+    ...    \ \ cpuManagerReconcilePeriod: 5s
+
+    Upload MicroShift Config    ${kubelet_config}
+
+Configure crio For Workload Partitioning
+    [Arguments]    ${cpus}
+    ${crioConfiguration}=    CATENATE    SEPARATOR=\n
+    ...    [crio.runtime]
+    ...    infra_ctr_cpuset = "${cpus}"
+
+    ...    [crio.runtime.workloads.management]
+    ...    activation_annotation = "target.workload.openshift.io/management"
+    ...    annotation_prefix = "resources.workload.openshift.io"
+    ...    resources = { "cpushares" = 0, "cpuset" = "${cpus}" }
+
+    Upload String To File    ${crioConfiguration}    ${crioConfigFile}
+
+Configure CPUAffinity In Systemd
```

Comment:
found that when creating systemd dropins using  RF framework with sudo and daemon +service restart  
selinux complains:
```
microshift-local setroubleshoot[30404]: SELinux is preventing systemd from open access on the file /etc/systemd/system/microshift.service.d/microshift-cpuaffinity.conf.
```

Reply:
handled with adding `-Z` when creating files and folders.